js/makeTable.js

```diff
@@ -0,0 +1,123 @@
+function makeTable() {
+	var data, sort_by, filter_cols; // Customizable variables
+	
+	var table; // A reference to the main DataTable object
+	
+	// This is a custom event dispatcher.
+	var dispatcher = d3.dispatch('highlight', 'select');
+	
+	// Main function, where the actual plotting takes place.
+	function _table(targetDiv) {
+	  // Create and select table skeleton
+	  var tableSelect = targetDiv.append("table")
+	    .attr("class", "display compact")
+			// Generally, hard-coding Ids is wrong, because then 
+			// you can't have 2 table plots in one page (both will have the same id).
+			// I will leave it for now for simplicity. TODO: remove hard-coded id.
+	    .attr("id", "gene_table") 
+	    .style("visibility", "hidden"); // Hide table until style loads;
+			
+	  // Set column names
+	  var colnames = Object.keys(data[0]);
+		if(typeof filter_cols !== 'undefined'){
+			// If we have filtered cols, remove them.
+			colnames = colnames.filter(function (e) {
+				// An index of -1 indicate an element is not in the array.
+				// If the col_name can't be found in the filter_col array, retain it.
+				return filter_cols.indexOf(e) < 0;
+			});
+		}
+		
+		// Here I initialize the table and head only. 
+		// I will let DataTables handle the table body.
+	  var headSelect = tableSelect.append("thead");
+	  headSelect.append("tr")
+	    .selectAll('td')
+	    .data(colnames).enter()
+		    .append('td')
+		    .html(function(d) { return d; });
+	
+		if(typeof sort_by !== 'undefined'){
+			// if we have a sort_by column, format it according to datatables.
+			sort_by[0] = colnames.indexOf(sort_by[0]); //colname to col idx
+			sort_by = [sort_by]; //wrap it in an array
+		}
+		
+
+	  // Apply DataTable formatting: https://www.datatables.net/
+	  $(document).ready(function() {
```

Comment:
I'm curious where we want to use a DOMReady function, shouldn't this be handled by databinding or some other event trigger?